### PR TITLE
Fix cropping for wrapped images

### PR DESCRIPTION
**Pull request by maarzt**
Within the legacy user interface ij.plugin.Resizer is used to crop an image.
This currently produces wrap results.
This PR adds a functional implementation of AbstractVirtualStack#deleteSlice
to fix this issue. (Only the deletion of first and last slice is supported currently)

### CODE REVIEW

File: src/main/java/net/imglib2/img/display/imagej/AbstractVirtualStack.java [hunk 1]
```diff
@@ -120,19 +141,24 @@ public void addSlice( final String sliceLabel, final ImageProcessor ip, final in
...
+			deleteFirtSlice();
+		else if ( n == size )
+			deleteLastSlice();
+		else
+			throw new UnsupportedOperationException( "AbstractVirtualStack only supports to delete first or last slice." );
 	}
 
-	@Override
-	public void deleteLastSlice()
+	private void deleteFirtSlice()
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/AbstractVirtualStack.java]: Typo Firt --> First

File: src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java [hunk 2]
```diff
@@ -150,14 +150,14 @@ public boolean isWritable()
 	}
 
 	@Override
-	public Object getPixels( final int n )
+	public Object getPixelsZeroBasedIndex( final int index )
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java]: I think this should be `protected`?

File: src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java [hunk 3]
```diff
@@ -150,14 +150,14 @@ public boolean isWritable()
...
 	{
-		final ArrayImg< T, ? > img = getSlice( n );
+		final ArrayImg< T, ? > img = getSlice( index );
 		return ( ( ArrayDataAccess< ? > ) img.update( null ) ).getCurrentStorageArray();
 	}
 
 	@Override
-	public void setPixels( final Object pixels, final int n )
+	public void setPixelsZeroBasedIndex( final int index, final Object pixels )
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java]: I think this should be protected?

File: src/main/java/net/imglib2/img/display/imagej/PlanarImgToVirtualStack.java [hunk 4]
```diff
@@ -148,9 +152,33 @@ private static int initSize( final Interval interval )
...
+	{
+		setPlaneCastType( img, indexer.applyAsInt( index ), wrapPixelsToAcces( pixels ) );
+	}
+
+	private static < A extends ArrayDataAccess< ? > > void setPlaneCastType( PlanarAccess< A > img, int index, ArrayDataAccess< ? > arrayDataAccess )
+	{
+		img.setPlane( index, ( A ) arrayDataAccess );
+	}
+
+	private ArrayDataAccess wrapPixelsToAcces( Object pixels )
```
Comment (tpietzsch) [inline src/main/java/net/imglib2/img/display/imagej/PlanarImgToVirtualStack.java]: Typo. `Acces` --> `Access`

**Comment by tpietzsch** [comment]
Please fix stuff that I commented.

This PR also adds implementation of `setPixels`. Could you comment on that? Is this also related to making `ij.plugin.Resizer` work?

**Comment by maarzt** [comment]
@tpietzsch Thanks for reviewing this. Looks like I had a bad typing day. I will fix it.

**Comment by maarzt** [comment]
> This PR also adds implementation of setPixels. Could you comment on that? Is this also related to making ij.plugin.Resizer work?

No these changes are not required to make ij.plugin.Resizer work.

In `AbstractVirtualStack` the `setPixels` method is now implemented, to make it easier for derived classes to correctly provide the `setPixels` functionality. It only has an effect, if the derived class implements `setPixelsZeroBasedIndex`.

The `PlanarImgToVirtualStack` class now implements exectly this `setPixelsZeroBasedIndex` method. This is a new feature. I split the PR into two commits such that this becomes more clear.

**Comment by maarzt** [comment]
@tpietzsch I did all the changes you requested

**Comment by tpietzsch** [comment]
👍

**Pull request merged**